### ISSUE TITLE
Delete all messages in a Discord channel

### ISSUE THREAD

**Issue by Paket236** [yourWaifu/sleepy-discord#148]
Is it possible to delete all messages from a Discord channel ?

**Comment by yourWaifu** [comment]
yes with bulk delete, however, it's going to take a while to get all the message ids

**Comment by Paket236** [comment]
Ok, I can get the lastMessageID, but how can I get IDs of previous messages?

**Comment by yourWaifu** [comment]
use getMessages, try not to spam it tho.

**Comment by yourWaifu** [comment]
Also can't you just delete a channel and recreate?

**Comment by Paket236** [comment]
Thanks, now I figured it out! However, on the run() function, an exception of type SleepyDiscord::ErrorCode with code 5005 is sometimes generated (if spamming), so I had to add exception handling and wrap it in a loop, as shown below. I hope so normal.
```
	thread t1( [&client]() {
		while(true) {
			try {
				client.run();
			}  catch (SleepyDiscord::ErrorCode i) {}
		}
	});
	t1.detach();
```
